examples/proposed/url-shortener.w

```diff
@@ -0,0 +1,141 @@
+bring cloud;
+
+let EMPTY_JSON = Json "empty";
+
+class Utils {
+  extern "./url-shortener.js" inflight makeId(): str;
+  extern "./url-shortener.js" inflight fetch(url: str, method: str, body: Json?): str;
+}
+let utils = new Utils();
+
+class UrlShortener {
+  urlLookup: cloud.Bucket;
+  idLookup: cloud.Bucket;
+
+  init() {
+    // map from urls to ids
+    this.urlLookup = new cloud.Bucket() as "UrlLookup";
+
+    // map from ids to urls
+    this.idLookup = new cloud.Bucket() as "IdLookup";
+  }
+
+  // Returns an id for the given long url. Creates a new id if one does not
+  // already exist.
+  inflight getId(url: str): str {
+    if this.urlLookup.exists(url) {
+      return this.urlLookup.get(url);
+    }
+  
+    let id = utils.makeId();
+    this.idLookup.put(id, url);
+    this.urlLookup.put(url, id);
+    return id;
+  }
+
+  // Get the url for the given id. Returns nil if the url does not have a
+  // corresponding id.
+  inflight getUrl(id: str): str? {
+    if this.idLookup.exists(id) {
+      return this.idLookup.get(id);
+    } else {
+      return nil;
+    }
+  }
+}
+
+class UrlShortenerApi {
+  api: cloud.Api;
+  shortener: UrlShortener;
+
+  init(shortener: UrlShortener) {
+    this.shortener = shortener;
+    this.api = new cloud.Api();
+
+    this.api.post("/create", inflight (req: cloud.ApiRequest): cloud.ApiResponse => {
+      let var requestUrl = "";
+      try {
+        // TODO: use `x ?? throw()` syntax
+        requestUrl = str.fromJson((req.body ?? EMPTY_JSON).get("url"));
```

Comment:
Could you use `Json {}` instead of `Json "empty"`?

Reply:
Why not `if let`?

Reply:
Ah I wrote this a while before we had `if let`, let me try that